ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1260,6 +1290,89 @@ private void testHardDeletePauseResume(boolean reloadIndex) throws InterruptedEx
     state.verifyEntriesForHardDeletes(idsDeleted);
   }
 
+  // getLogSegmentsNotInJournal helpers
+
+  /**
+   * Tests {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   * @param maxLogSegmentsToBeIgnored number of log segments not to be returned via {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   * @param maxEntriesInJournal max number of entries in {@link Journal}
+   * @param reloadIndex {@code true} if index needs to be reloaded and tested for {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   *                    @@code false} otherwise
+   * @throws InterruptedException
+   * @throws StoreException
+   * @throws IOException
+   */
+  private void testGetLogSegmentsNotInJournal(int maxLogSegmentsToBeIgnored, int maxEntriesInJournal,
```

Comment:
If 1 and 2 segments are all that we will test, I think we should just drop this function (neither `maxLogSegmentsToBeIgnored` nor `maxEntriesInJournal` is really variable and this function works only in particular cases) and write all this code in `getLogSegmentsNotInJournalTest()`. Also I don't see why a `withReload()` has to be a different test. Why not just do it in `getLogSegmentsNotInJournalTest()`?